### ISSUE TITLE
Serialize/Deserialize a Peripheral instance for a faster connection. 

### ISSUE THREAD

**Issue by janleskovec** [Polidea/FlutterBleLib#454]
My current implementation to connect to a specific BLE device is to search for the device by it's name and then connect to it. Is it possible to perform the "discovery" procedure only the first time and then serialize the `Peripheral` object that is generated and connect to it multiple times?

**Comment by mikolak** [comment]
Hi!

By "discovery" here you mean scanning for peripherals, right? For now it is not possible, but I'll be adding a method to obtain a peripheral from its id (UUID on iOS, MAC address on Android). I'm unable to give you a timeline though.

**Comment by slole** [comment]
Yes, that is exactly what I meant. And that is the exact feature I am looking for. Thank you. No worries on the timing, I was just wondering whether it is possible or is there an underlying limitation.
